clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2415,27 +2380,79 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::pick_next_input(output_ordinal_t
     }
     // We can't easily place these stats inside set_cur_input() as we call that to
     // temporarily give up our input.
-    update_switch_stats(output, prev_index, index);
+    on_context_switch(output, prev_index, index);
     set_cur_input(output, index);
     return res;
 }
 
 template <typename RecordType, typename ReaderType>
 void
-scheduler_impl_tmpl_t<RecordType, ReaderType>::update_switch_stats(
+scheduler_impl_tmpl_t<RecordType, ReaderType>::on_context_switch(
     output_ordinal_t output, input_ordinal_t prev_input, input_ordinal_t new_input)
 {
+    bool do_inject_switch_seq = false;
     if (prev_input == new_input)
         ++outputs_[output].stats[memtrace_stream_t::SCHED_STAT_SWITCH_NOP];
     else if (prev_input != sched_type_t::INVALID_INPUT_ORDINAL &&
-             new_input != sched_type_t::INVALID_INPUT_ORDINAL)
+             new_input != sched_type_t::INVALID_INPUT_ORDINAL) {
         ++outputs_[output].stats[memtrace_stream_t::SCHED_STAT_SWITCH_INPUT_TO_INPUT];
-    else if (new_input == sched_type_t::INVALID_INPUT_ORDINAL)
+        do_inject_switch_seq = true;
+    } else if (new_input == sched_type_t::INVALID_INPUT_ORDINAL)
```

Comment:
> Now, in the switch insertion offline test, we have 11 context switch injections (== input-to-input + idle-to-input switches), 

Hmm, what about input-to-idle?  I think even if we do agree to do nothing, as this PR does, there should be a comment here (and in the PR description) explaining why.  We're going to transition to the kernel idle loop so we are in a different context.  The idle loop doesn't do much: so there's presumably less antagonistic code.  For the register switching code and barriers: not sure: do we know there are no barriers, or other important components that simulators want to see, on a switch to idle?  If we don't know: let's put a XXX comment acknowledging we might want some kind of kernel sequence here but we're not sure what yet.

Reply:
My thinking was similar to what we do for "total context switches". See the following comment in schedule_stats_t::get_scheduler_stats:

```
    // XXX: Currently, schedule_stats is measuring swap-ins to a real input.  If we
    // want to match what "perf" targeting this app would record, which is swap-outs,
    // we should remove idle-to-input and add input-to-idle (though generally those
    // two counts are pretty similar).  OTOH, if we want to match what "perf"
    // systemwide would record, we would want to add input-to-idle on top of what we
    // have today.
```

> let's put a XXX comment acknowledging we might want some kind of kernel sequence here but we're not sure what yet.

Yes, done.